### ISSUE TITLE
UnusedLocalVariable reported for autocloseable who's only actions are performed in .close

### ISSUE THREAD

**Issue by mebigfatguy** [pmd/pmd#6216]
**Affects PMD Version:** 
7.17.0

**Rule:**
UnusedLocalVariable

Please provide the rule name and a link to the rule documentation:
[<https://docs.pmd-code.org/latest/pmd_rules_XXX_XXX.html#XXX>](https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_bestpractices.html#unusedlocalvariable)

**Description:**

When an autocloseable is _ONLY_ used by what happens in the .close (such as)

**Code Sample demonstrating the issue:**

```java

try (AutoCloseable lock = Locker.getGlobalLock()) {

        // do some stuff unrelated to the lock variable here
} 

```
getLock creates some sort of global lock, and the .close() unlocks the lock.

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
ant

**Comment by pzygielo** [comment]
- https://github.com/pmd/pmd/issues/5747?

**Comment by mebigfatguy** [comment]
ah.. sorry.
